File: quote/app/controllers/Application.scala
package controllers

import models.Quote
import play.api.mvc.{Controller, Action}
import play.api.libs.json._
import views.html.index

object Application extends Controller {

  val getIndex = Action { request =>
    Ok(index.render())
  }

  val getAQuote = Action { request =>
    val quote = Quote.getRandomQuote
    Ok(Json.toJson(JsObject(Seq("quote" -> JsString(quote.text + " - " + quote.author)))))
  }
}

File: originator/app/controllers/Application.java
package controllers;

import play.mvc.Controller;
import play.mvc.Result;
import views.html.index;
import views.html.requestButton;

public class Application extends Controller {

    public static Result index() {
      return ok(index.render("Orginator says hello!"));
    }

    public static Result postRequestButton() {
        return ok(requestButton.render("Push the button"));
    }

}
